packages/contractkit/src/utils/tx-result.ts

```diff
@@ -0,0 +1,59 @@
+import debugFactory from 'debug'
+import { Future } from 'src/utils/future'
+import PromiEvent from 'web3/promiEvent'
+import { TransactionReceipt } from 'web3/types'
+
+const debug = debugFactory('contractkit:txresult')
+
+export function toTxResult(pe: PromiEvent<any>) {
+  return new TransactionResult(pe)
+}
+
+export class TransactionResult {
+  private hashFuture = new Future<string>()
```

Comment:
Could you elaborate on why you chose Future's here over the build in promise? IMO the latter are more familiar to most developers?